test/inductor/test_templated_attention.py

```diff
@@ -371,6 +376,43 @@ def func(q, k, v, score_mod):
         # Ensure that two kernels are generated
         FileCheck().check_count(".run(", 2, True).run(code[0])
 
+    @supported_platform
+    @common_utils.parametrize("score_mod", [_identity, _causal, _times_two, _squared])
+    def test_aot_eager_gradcheck(self, score_mod):
+        make_tensor = functools.partial(
+            torch.randn,
+            (2, 2, 8, 4),
+            device="cuda",
+            dtype=torch.float64,
+            requires_grad=True,
+        )
+        query, key, value = make_tensor(), make_tensor(), make_tensor()
+
+        func = torch.compile(_templated_attention, backend="aot_eager", fullgraph=True)
+
+        self.assertTrue(torch.autograd.gradcheck(func, (query, key, value, score_mod)))
+
+    @supported_platform
+    def test_captured_score_mod_aot_eager_gradcheck(self):
+        make_tensor = functools.partial(
+            torch.randn,
+            (2, 2, 8, 4),
+            device="cuda",
+            dtype=torch.float64,
+            requires_grad=True,
+        )
+        query, key, value = make_tensor(), make_tensor(), make_tensor()
+
+        func = torch.compile(_templated_attention, backend="aot_eager", fullgraph=True)
+
+        head_offset = torch.rand(H, device="cuda", dtype=torch.float64)
+
+        # Uses a lifted variable and creates correct joint graph is required for gradcheck
+        def score_mod(score, b, h, m, n):
+            return score * index(head_offset, [h])
+
+        self.assertTrue(torch.autograd.gradcheck(func, (query, key, value, score_mod)))
```

Comment:
use gradcheck with "raise_exception=True": so the error messages are less opaque